lib/meilisearch/index.rb

```diff
@@ -393,5 +393,19 @@ def update_pagination(pagination)
     def reset_pagination
       http_delete "/indexes/#{@uid}/settings/pagination"
     end
+
+    def typo_tolerance
+      http_get("/indexes/#{@uid}/settings/typo-tolerance")
+    end
+    alias get_typo_tolerance typo_tolerance
+
+    def update_typo_tolerance(typo_tolerance_attributes)
+      http_patch("/indexes/#{@uid}/settings/typo-tolerance", typo_tolerance_attributes)
```

Comment:
A good addition here is to use the utility method to transform everything in camel case, so we can use snake_case in our ruby/rails apps :)

It should look like this:

```rb
    def update_typo_tolerance(typo_tolerance_attributes)
      attributes = Utils.transform_attributes(typo_tolerance_attributes)

      http_patch("/indexes/#{@uid}/settings/typo-tolerance", attributes)
    end
```